File: asakatsu/20221010/4.py

# https://atcoder.jp/contests/arc148/tasks/arc148_b

N = int(input())
S = list(input())

# 区間を選び、反転して更新
ans = S[:]

for i in range(N):
    for j in range(i+1, N):
        tmp = S[:]
        tmp[i:j] = ["d" if c=="p" else "p" for c in reversed(tmp[i:j])]
        
        # 更新
        if ans > tmp:
            ans = tmp

print("".join(ans))

File: asakatsu/asa20221010/_4.rs
// https://atcoder.jp/contests/arc148/tasks/arc148_b

/*

## 方針
文字列中の区間`S[i..j]`を反転させたときの

 */

fn main() {
    let n: usize = {
        let mut s = String::new();
        std::io::stdin().read_line(&mut s).unwrap();
        s.trim().parse().unwrap()
    };
    let s: String = {
        let mut s = String::new();
        std::io::stdin().read_line(&mut s).unwrap();
        s.trim().to_string()
    };

    const INF: isize = 1e18 as isize;
    let mut rev = [[INF; 5050]; 5050];
}
